src/cli/commands/test/mod.rs

```diff
@@ -151,15 +153,20 @@ pub(crate) fn test_single_entrypoint(
 	let mut output = String::new();
 	let execution_uuid = Uuid::new_v4();
 	init_buffer(execution_uuid);
-	let res_cairo_run = cairo_run(
-		&path_to_compiled,
-		&test_entrypoint,
-		false,
-		false,
-		hint_processor,
-		execution_uuid,
-		hooks,
-	);
+
+	let program = match Program::from_json(program, &test_entrypoint) {
+		Ok(program) => program,
+		Err(e) => {
+			output.push_str(&format!(
+				"[{}] {}\nError: failed to deserialize program",
```

Comment:
Not a good error description. The two errors `deserialize_from_program_json` can return are:
- `Err(ProgramError::EntrypointNotFound(entrypoint.to_string()))`
- `ProgramError::ConstWithoutValue(key.to_owned()))`

The error is more something like `Invalid program`

And we prefer messages as follow `<MyCustomMessageError>: <TheSourceError>`
(check it at other places in your PR)